src/xdg-autostart-generator/xdg-autostart-generator.c

```diff
@@ -0,0 +1,771 @@
+                        return 0;
+                }
+        }
+
+        r = format_exec_start(service->exec_string, &exec_start);
+        if (r < 0) {
+                log_info("Not generating service for XDG autostart %s, Exec line is invalid",
+                         service->name);
+                return 0;
+        }
+
+        /* Partially handle AutostartCondition (as implemented by GNOME)
+         *
+         * Not implemented are:
+         *  - GSettings (may be updated at runtime)
+         *  - GNOME3 if-session/unless-session (which seems unused and undocumented)
+         */
+        if (service->autostart_condition) {
+                const char *kind;
+                const char *key;
+                const char *state;
+                size_t kind_length;
+                _unused_ size_t key_length;
+
+                /* Split into first word and rest (we don't use key_length). */
+                state = service->autostart_condition;
+                kind = split(&state, &kind_length, WHITESPACE, 0);
+                key = split(&state, &key_length, WHITESPACE, 0);
+                printf("kind: %p\n", kind);
+                printf("key: %s\n", key);
+
+                if (key && (strncaseeq("if-exists", kind, kind_length) || strncaseeq("unless-exists", kind, kind_length))) {
+                        _cleanup_free_ char *e = NULL;
+
+                        /* The key is a filename, just escape it. */
+                        e = cescape(key);
+                        asprintf(&condition_string,
+                                 "ConditionPath=%s%%E/%s\n",
+                                 strncaseeq("if-exists", kind, kind_length) ? "" : "!",
+                                 e);
+                        if (!condition_string)
+                                return log_oom();
+                } else {
+                        /* Unsupported or something is wrong, we should never run this service */
+                        log_warning("Not generating service for XDG autostart %s with unsupported AutostartCondition", service->name);
+                        return 0;
+                }
+        }
+
+        path_escaped = specifier_escape(service->path);
+        if (!path_escaped)
+                return log_oom();
+
+        unit = prefix_roota(arg_dest, service->name);
+
+        f = fopen(unit, "wxe");
+        if (!f)
+                return log_error_errno(errno, "Failed to create unit file %s: %m", unit);
+
+        fprintf(f,
+                "# Automatically generated by systemd-xdg-autostart-generator\n\n"
+                "[Unit]\n"
+                "Documentation=man:systemd-xdg-autostart-generator(8)\n"
+                "SourcePath=%s\n"
+                /* Disallow manual starting to make Conflicts= safer. */
+                "RefuseManualStart=yes\n"
+                "PartOf=graphical-session.target\n"
+                "Requisite=xdg-desktop.target\n",
```

Comment:
If systemd is providing the generator. I think systemd should be shipping this target in its default units

Reply:
Added, documentation is still missing though.